sklearn/metrics/_ranking.py

```diff
@@ -1616,11 +1618,26 @@ def ndcg_score(y_true, y_score, *, k=None, sample_weight=None, ignore_ties=False
     ...           scores, k=1, ignore_ties=True)
     0.5
     """
+
     y_true = check_array(y_true, ensure_2d=False)
     y_score = check_array(y_score, ensure_2d=False)
     check_consistent_length(y_true, y_score, sample_weight)
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
+
+    if (isinstance(y_true, np.ndarray)):
+        if (y_true.min() < 0):
+            warnings.warn(
+                "ndcg_score should not use negative y_true values",
+                DeprecationWarning,
+            )
+    else:
+        for value in y_true:
+            if (value < 0):
+                warnings.warn(
+                    "ndcg_score should not use negative y_true values",
+                    DeprecationWarning,
+                )
```

Comment:
The documentation states that  `y_true : ndarray of shape (n_samples, n_labels)` so do we really need this clause?

Reply:
This clause was for one of the tests that from it seemed did not use numpy arrays, but the regular one. I found that removing the else clause, causes these kinds of tests to fail. Unless I am misunderstanding the issue, it is necessary for some of tests to pass